### PR TITLE
[Story] Select issues and PRs and move them to Up Next in planning (#283)

### LINKED ISSUE #283: [Story] Select issues and PRs from across repositories and move them to Up Next in one planning session
## Description
Iteration Planning lets the user select issues and PRs from included repos and move them to Up Next with Focus Order.

## User Story
**As a** solo developer,
**I want** to select issues and PRs from across repositories and move them to Up Next in one planning session,
**So that** my work queue is populated quickly and intentionally.

## Acceptance Criteria
- [ ] Route `/pm-workflow/planning` with candidate picker and current Up Next list.
- [ ] Add uses Projects v2 Status Up Next (matched by option name, case-insensitive) and sequential Focus Order for stories/enablers/tests only.
- [ ] Feature/Epic cards skip Focus Order.
- [ ] Items not yet on the board are added then updated (reuse Triage add-item flow).
- [ ] Success and failure snackbars; inaccessible board warning.
- [ ] Unit tests for Focus Order sequencing.

## Related Epic / Milestone
Feature #272 — Cross-Repo PM Workflow. Milestone v1.1.0.

## Implementation References
- **Wireframe:** plan/wireframes/pm-workflow-wireframe.md
- **Parent issue:** #272
- **Test issue:** #387
- **Related decisions:** DEC-027, DEC-028, DEC-029
- **Feature plan doc:** plan/cross-repo-pm-workflow-project-plan.md

## Additional Context
- **Blocked by:** #382, #383, #384
- **Blocks:** #284, #285, #286

### PR DESCRIPTION
**Pull request by markheydon**
## Summary of Changes

Implements the Iteration Planning slice for issue #283: `/pm-workflow/planning` now shows the current Up Next batch and a searchable candidate picker across included repositories. Adding a candidate moves it to Up Next on the selected Projects v2 board (Status matched case-insensitively), creates a board card when needed via the Triage add-item flow, and assigns sequential Focus Order for stories, enablers, and tests while skipping Feature and Epic cards.

## Related Issue(s)

Closes #283

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ Feature (non-breaking change that adds functionality)
- [ ] 🔧 Chore / maintenance (dependency update, refactoring, tooling)
- [ ] ♻️ Refactoring (no functional change, improves code quality)
- [ ] 📝 Documentation (updates to docs, comments, or README)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes follow the coding conventions in `AGENTS.md`
- [x] All new and existing tests pass (`dotnet test`)
- [x] I have added or updated tests to cover my changes
- [x] I have updated relevant documentation in `docs/` or `plan/`
- [x] No new compiler warnings have been introduced
- [x] All text in comments, strings, and docs is written in **UK English**
- [x] If this adds a new feature, a GitHub Issue exists (or is updated), Project #8 is synced, and end-user docs under `website/` (or developer docs under `docs/`) have been updated

## Screenshots (if applicable)

Not captured in this PR. Planning UI follows the wireframe; docs-capture screenshots remain for the broader PM Workflow publication gate (#387).

## Additional Notes

- Application layer: `IIterationPlanningService`, `PlanningFocusOrderSequencer`, and view mapper DTOs.
- UI: `PmWorkflowPlanningPanel` with success and failure snackbars; inaccessible-board warning reuses shared PM Workflow chrome.
- Tests: 20 new unit/component tests covering Focus Order sequencing, add-to-Up-Next writes, and planning page shell.
- Out of scope for #283 (follow-up issues): capacity soft cap (#284), stalled Up Next gate (#285), bulk milestone (#286).

### CODE REVIEW

File: src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                .ConfigureAwait(false);
+
+        await _gitHubService
+            .UpdateProjectBoardItemStatusAsync(
+                projectId,
+                projectItemId,
+                catalogue.FieldIds.StatusFieldId,
+                upNextOption.OptionId,
+                cancellationToken)
+            .ConfigureAwait(false);
```
Comment (markheydon) [inline src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs]: **High — stale scoped catalogue after status update.** `UpdateProjectBoardItemStatusAsync` is called directly on `IGitHubService`, so `ProjectItemCatalogueService` keeps serving the pre-mutation catalogue until `UpdateFocusOrderAsync` runs. That invalidation only happens for story/enabler/test labels; Feature/Epic adds skip Focus Order entirely (#283 AC).

The panel reloads immediately after a successful add (`LoadPlanningViewAsync` → `GetCatalogueAsync`), so users can see a success snackbar while the item still appears in the candidate picker and is missing from Up Next.

**Suggested fix:** invalidate the project catalogue after every successful board mutation (status and add-item), e.g. add `InvalidateCatalogue(string projectId)` on `IProjectItemCatalogueService` and call it after the status update (and after triage add when the cache cannot reflect the new card). Add a service test for the Feature/Epic path that asserts a subsequent `GetCatalogueAsync` refetches.

**Review by markheydon — COMMENTED** [review]
## Code review summary

Solid slice for #283: layered `IIterationPlanningService`, focused mapper/sequencer helpers, MudBlazor panel with good `data-testid` coverage, and 20 targeted unit/component tests. Architecture, UK English, and xUnit/NSubstitute conventions look good. CI (Build and Test, Playwright) is green on this revision.

## Findings

### High

1. **Stale project-board catalogue after writes that skip Focus Order** — see inline comment on `IterationPlanningService.cs`. Feature/Epic adds (and any path that does not call `UpdateFocusOrderAsync`) leave `ProjectItemCatalogueService`'s scoped cache populated with pre-mutation data. The panel immediately reloads via `GetPlanningViewAsync`, so the success snackbar can show while Up Next and the candidate list remain stale until the DI scope ends.

### Medium

2. **User guide still documents Planning as a placeholder** — `website/content/docs/pm-workflow.md` (lines 8, 27, 138, 237) still says Iteration Planning is not shipped. The PR checklist correctly leaves docs unchecked and defers publication to #387; either update the guide in this PR or track that explicitly on #387 before calling #283 done.

3. **No Playwright shell coverage for `/pm-workflow/planning`** — `tests/E2E/tests/pm-workflow.spec.ts` and `tests/E2E/fixtures/accessibility.ts` cover Daily Focus, Backlog, and Repos but not Planning. bUnit tests help, but repo alignment rules expect a planning-route assertion (even no-board/empty chrome) alongside the guide update.

### Low

4. **`CancellationToken.None` for `AddToUpNextAsync`** — `PmWorkflowPlanningPanel` already owns a `viewLoadCts` for loads; reusing a dedicated add-operation CTS (or linking to component disposal) would let navigation/disposal cancel in-flight GitHub writes.

## Merge recommendation

**Request changes** — fix the catalogue cache invalidation (High) before merge. Medium items can follow on #387 if that is the agreed publication gate, but should not be forgotten.

*(GitHub blocks self-request-changes on this PR; recommendation is recorded as COMMENT.)*

File: src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs (hunk 1 above)
Comment (markheydon) [reply src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs]: Addressed in 2ab17b7: added `IProjectItemCatalogueService.InvalidateCatalogue` and call it immediately after a successful status update in `IterationPlanningService.AddToUpNextAsync`. Added `ProjectItemCatalogueServiceTests.InvalidateCatalogue_AfterCatalogueLoad_ForcesRefetchOnNextGet` and asserted `InvalidateCatalogue` on the Feature-label service test.

**Comment by markheydon** [comment]
## Review comment loop — addressed

All findings from the code review have been implemented in `2ab17b7`.

| Severity | Finding | Resolution |
|----------|---------|------------|
| High | Stale `ProjectItemCatalogueService` cache after Feature/Epic adds | Added `InvalidateCatalogue(projectId)` and call it after every successful Up Next status write; service tests cover invalidation and Feature-label path |
| Medium | User guide still marked Planning as placeholder | Updated `website/content/docs/pm-workflow.md` with partial Iteration Planning section and availability table |
| Medium | No Playwright shell for `/pm-workflow/planning` | Added `Iteration Planning` describe in `pm-workflow.spec.ts`, accessibility route, and `USER_DOCS_ALIGNMENT.md` mapping |
| Low | `CancellationToken.None` on add | `PmWorkflowPlanningPanel` now uses a dedicated `addToUpNextCts` cancelled on dispose |

Scoped tests: 36 passed (`IterationPlanning*`, `PlanningFocusOrder*`, `PmWorkflowPlanning*`, `ProjectItemCatalogueService*`).

File: src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        _projectItemCatalogueService.InvalidateCatalogue(projectId);
+
+        var assignsFocusOrder = PlanningFocusOrderSequencer.ShouldAssignFocusOrder(labels);
+        double? focusOrderAssigned = null;
+
+        if (assignsFocusOrder)
+        {
+            if (string.IsNullOrWhiteSpace(catalogue.FieldIds.FocusOrderFieldId))
+            {
+                throw new InvalidOperationException("The project board does not expose a Focus Order field.");
```
Comment (markheydon) [inline src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs]: **High — contradicts UI and user guide.** When `assignsFocusOrder` is true but the board has no Focus Order field, this throws *after* `UpdateProjectBoardItemStatusAsync` succeeds (line 102–109). The panel warning explicitly says story/enabler/test cards **can still move to Up Next** without Focus Order (`pm-workflow-planning-no-focus-order-field`), and the user guide matches that intent.

Suggested fix: treat a missing field like Feature/Epic skip — return success with `FocusOrderSkipped: true` (and a user-facing skip reason such as "Focus Order unavailable on this board") instead of throwing. Add a service test for `type/story` + empty `FocusOrderFieldId` asserting Status is set and no exception is thrown.

**Review by markheydon — COMMENTED** [review]
## Code review summary

Strong slice for #283 after the follow-up commits: layered `IIterationPlanningService`, focused mapper/sequencer helpers, MudBlazor panel with good `data-testid` coverage, catalogue invalidation after Status writes, tab caching via `PmWorkflowChromeCoordinator`, and targeted unit/component/E2E tests. Architecture, UK English, xUnit/NSubstitute conventions, and PR metadata all look good. CI is green on this revision.

Prior review findings (catalogue cache invalidation, Planning Playwright shell, user-guide Planning section, add-operation cancellation) appear addressed.

## Findings

### High

1. **Story/enabler/test adds fail on boards without a Focus Order field** — see inline comment on `IterationPlanningService.cs`. Status is already written to Up Next before the throw, so the UI shows an error snackbar even though the panel copy says those cards *can still move to Up Next* when Focus Order is unavailable.

### Medium

2. **Repo Management guide still says Planning exclusions are future work** — `website/content/docs/pm-workflow.md` lines 208 and 227 still say Planning candidates will honour exclusions *when that panel ships*, but Planning already loads candidates through `PmWorkItemCatalogueService`, which applies `ExcludedRepositories` today.

3. **Planning filter control mismatch in the user guide** — the Iteration Planning section (line 174) describes **Issues** and **Pull requests** checkboxes, but the shipped UI uses a **Type** dropdown (`All types` / `Issues` / `Pull requests`).

### Low

4. **PR checklist not updated** — the template still leaves documentation items unchecked even though this revision updates `website/content/docs/pm-workflow.md`, `tests/E2E/USER_DOCS_ALIGNMENT.md`, and Planning Playwright coverage.

## Merge recommendation

**Request changes** — fix the missing Focus Order field path (High) before merge. Medium doc nits can land in this PR or a quick follow-up before #283 is closed.

File: src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs (hunk 2 above)
Comment (markheydon) [reply src/Application/SoloDevBoard.Application/Services/PmWorkflow/IterationPlanningService.cs]: Fixed in d316ea6: story/enabler/test adds now skip Focus Order when the board has no field instead of throwing after Status is written. Added `FocusOrderUnavailableMessage`, service and sequencer tests, and snackbar/table skip-reason wiring via `DescribeFocusOrderSkipReason(..., hasFocusOrderField)`.

**Comment by markheydon** [comment]
## Review comment loop — addressed in d316ea6

All findings from the latest code review have been implemented:

| Finding | Action |
|---------|--------|
| **High** — story/enabler/test adds throw when Focus Order field is missing | `IterationPlanningService` now skips Focus Order assignment and returns `FocusOrderSkipped: true`; added `FocusOrderUnavailableMessage` and service/sequencer tests. |
| **Medium** — Repo Management guide still says Planning exclusions are future work | Updated `website/content/docs/pm-workflow.md` lines 208 and 227 to present-tense exclusion copy. |
| **Medium** — user guide describes checkboxes instead of Type dropdown | Updated Iteration Planning candidate picker docs to describe the **Type** dropdown. |
| **Low** — PR checklist not updated | Documentation checklist items ticked on this PR. |

Unresolved review thread on `IterationPlanningService.cs` has been replied to and resolved. Scoped Application tests for Iteration Planning pass locally.
